Evolutions et corrections 2.2 dont champs stabilité au feu en minutes

A511701 opened the pull request:
Evolutions :
- Ajout de la variable d'environnement PREVARISC_PARKINGS_TYPES précisant les Ids des types d'activités liés aux parkings
- Déplacement et agrandissement de l'identifiant externe
- Ajout de la possibilité de pas avoir un copie de travail git et forcer une version applicative dans le footer
- Fixes #302 Passage des champs de stabilité au feu et isolement en minutes au lieu des heures
- En lien avec #306, retrait de l'ajout du champ LIBELLE_RUETYPE dans l'affichage des adresses

Corrections :
- Correction du libellé "Places de stationnement" qui ne revient pas à Public lors de la sélection d'une autre activité
- Corrections de warnings (undefined indexes) sur la génération des documents
- Corrections du bouton modifier le dossier qui apparaît à la création du dossier

Commit:
Merge pull request #38 from SDIS62/master

Syncing with SDIS62 / prevarisc

Commit:
Ajout de la variable d'environnement PREVARISC_PARKINGS_TYPES précisant les Ids des types d'activités liés aux parkings
Correction du libellé "Places de stationnement" qui ne revient pas à Public lors de la sélection d'une autre activité

Commit:
Corrections de warnings (undefined indexes) sur la génération des documents
Corrections du bouton modifier le dossier qui apparaît à la création du dossier
Ajout de messages de debug sur l'ajout d'une pièces jointe

Commit:
Correction d'une coquille sur le SQL d'init sur la colonne de droit public

Commit:
Déplacement et agrandissement de l'identifiant externe
Ajout de la possibilité de pas avoir un copie de travail git et forcer une version applicative dans le footer

Commit:
Fixes #302 Passage des champs de stabilité au feu et isolement en minutes au lieu des heures
En lien avec #306, retrait de l'ajout du champ LIBELLE_RUETYPE dans l'affichage des adresses

kdubuc commented:
Pourquoi la variable PREVARISC_PARKINGS_TYPES ? J'ai poussé d'ailleurs une migration sur le type d'activité des parkings ici https://github.com/SDIS62/prevarisc/blob/2.x/sql/migrations/2.2-11-activitesPS.sql

A511701 commented:
Tout à fait, j'ai vu votre migration.
Il s'avère que le SDIS29 utilise d'autres types d'activité. Un travail a été fait pour variabiliser les ID à ce niveau.
Dans le fiche établissement, onglet principal, un JS change le libellé de l'effectif s'il s'agit d'un type parking, d'où l'évolution :)

kdubuc commented:
Bon je viens de faire une petite revue de code. Il y a pas mal de conflit, si vous pouvez gérer ça de votre côté en pullant la derniere révision de la branche 2.x et adapter vos corrections en conséquences, ça serait génial. Il y a plusieurs points qui vont rentrer en conflict :
- le composer.lock, vous n'utilisez pas les dernières versions des dépendances, un petit composer update de votre côté permettra d'effacer cette différence ;
- Il fallait que ça arrive; j'ai poussé une fichier de migration portant le même nom que le votre, on aurait du écrire une convention de nommage là dessus (timestamp+nom) par exemple .. Vous pouvez la renommer ? ;
- La vesrion prevarisc, j'ai supprimé les occurences de $this->version_prevarisc cette après midi, et ça tape un fichier ou vous avez ajouter la gestion de version via getenv ou git. Garder votre modification sur ce fichier ! Par contre, une seule variable d'environnement n'aurait pas suffit pour la version, plut^t qu'une pour la rev, la branche et la version ? Par exemple PREVARISC_VERSION = "2.2 (6560c49074b44930ee29f9dd2611f38ca57bf522)" si vous voulez quand même garder une trace du commit.
  Et voila, et c'est déjà pas mal ;)

kdubuc commented:
Justement, vous pouvez gérer ça plus simplement en ajoutant des activité en BD au type PS (id 16) ! Plus besoin de toucher au JS, lui il se base sur le type, et la liste associée au type PS sera en fonction de votre BD, et donc de la liste du SDIS !

A511701 commented:
Le problème est que le SDIS29 a d'autres types d'activités et qu'il souhaite les voir apparaître dans un certain ordre, d'où la création des types d'activités les uns après les autres pour suivre l'ordre des auto-increment car il n'y a pas de colonne "ordre" dans la table des types d'activités.

Mais en effet, cette solution était également faisable :)

Je vous pousse une modification qui adresse les points relevés.

kdubuc commented:
Vous pouvez jouer avec l'ID primaire pour l'ordre !
Okay ! :wink:

Commit:
Merge pull request #39 from SDIS62/2.x

Syncing with SDIS62 / prevarisc

A511701 commented:
Au sujet de la version et votre dernière remarque, je n'ai pas bien saisi : souhaitez-vous finalement garder le $this->version dans la classe View ? Ou finalement repasser cette logique dans le partial footer.phtml ?

Pour la révision, en effet, 3 variables font de trop pour cette notion. J'ai été obligé de faire comme ceci à cause du 

```
href="https://github.com/SDIS62/prevarisc/commit/<?php echo $this->revision_prevarisc ?>" 
```

qui nécessite d'avoir le numéro du commit séparé du numéro de la version...
2 variables du coup ?

A511701 commented:
Attention : la dépendance sur apigen/apigen résoud de nombreuses dépendances sur le framework nette...

kdubuc commented:
Non, on garde bien votre $this->version dans le plugin View. Et 2 variables c'est bien ! Ou une si on supprime le lien vers la version quand elle spécifiée en dur.

A511701 commented:
ok

Commit:
Merge branch 'master' of https://github.com/A511701/prevarisc into 2.2

Conflicts:
	application/plugins/View.php
	composer.lock

A511701 commented:
Normalement, le merge a été effectué.
Pouvez-vous contrôler et me dire si tout est OK ?

kdubuc commented:
Vous avez gardé PREVARISC_PARKINGS_TYPES ? ça serait bien de gérer ça côté BD plutot qu'une modification de code pour un cas particulier .. 
Sinon le merge est OK

A511701 commented:
Oui, cela laisse la possibilité à un SDIS d'utiliser d'autres types d'activités, c'est un plus ! :-)
Les ID en dur dans le code sont clairement à éviter... Dans ce cas, mieux vaut se baser sur le code activité : PS ?

Commit:
Correction de l'appel au service IGN de géolocalisation : envoi du code postal entier à la place des 2 premiers chiffres du code insee

Commit:
Correction : la recherche multi-critère incluant la recherche sur une commune rendait 0 résultats : la recherche imposait la présence d'une commune égale pour les sites et cellules liées

Commit:
Retrait de la variable PARKING_TYPE et ajout d'une fonction javascript surchargeable
Correction d'un cas à la marge sur les alertes : les établissements sans libellé

Commit:
Correction des simples quotes dans les groupements de communes qui cassent les listes des membres de commissions et services instructeurs

Commit:
Correction mineure sur le message lorsqu'il n'y a pas de commissions

kdubuc commented:
Pour profiter du futur merge de cette PR, pouvez vous généraliser le style d'affichage des dossiers dans la recherche à la partie des dossiers liés à un établissement? Au moins nous aurons un style unique de représentation d'une liste de dossier dans l'application.
Au fait, je ne vois plus la partie courrier dans la recherche, c'est normal ?

A511701 commented:
Nous avons retiré la partie courriers, car au final, c'était devenu en doublons avec les dossiers.

Pour le "style d'affichage des dossiers dans la recherche à la partie des dossiers", je ne vois pas de quoi il s'agit. Pouvez-vous préciser ?
